src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -237,12 +251,18 @@ RouterExecutorRun(QueryDesc *queryDesc, ScanDirection direction, long count, Tas
  * also generate warnings for individual placement failures.
  */
 static int32
-ExecuteDistributedModify(Task *task)
+ExecuteDistributedModify(Query *query, Task *task)
 {
 	int32 affectedTupleCount = -1;
 	ListCell *taskPlacementCell = NULL;
 	List *failedPlacementList = NIL;
 	ListCell *failedPlacementCell = NULL;
+	StringInfo queryString = makeStringInfo();
+
+	ExecuteFunctions(query);
```

Comment:
I think we should make this conditional. Remember during planning whether we need to evaluate functions. That way we don't continually have to redo that step unnecessarily.